File: app/restaurant/lib/src/models/restaurant.dart
import 'package:restaurant/src/models/address.dart';
import 'package:restaurant/src/models/location.dart';

class Restaurant {
  const Restaurant({
    required this.id,
    required this.name,
    required this.displayImgUrl,
    required this.type,
    required this.location,
    required this.address,
  });

  final String id;
  final String name;
  final dynamic displayImgUrl;
  final String type;
  final Location location;
  final Address address;
}

File: server/src/models/restaurant.ts
import { Address } from './address';
import { Location } from './location';

export default class Restaurant {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly type: string,
    public readonly rating: number,
    public readonly displayImageUrl: string,
    public readonly location: Location,
    public readonly address: Address
  ) {}
}
